### PR TITLE
Issue 4348 - Add tests for dsidm

### PR DESCRIPTION
**Pull request by bsimonova**
Description:
Created tests for dsidm user option and enhanced
the src/lib389/lib389/cli_idm/__init__.py and src/lib389/lib389/cli_base/__init__.py
so the output gets caught to topology LogCapture to compare the results.

Relates: https://github.com/389ds/389-ds-base/issues/4348

Reviewed by: ???

### CODE REVIEW

File: dirsrvtests/tests/suites/clu/dsidm_user_test.py [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    topology_st.logcap.flush()
+
+    log.info('Test dsidm user list without json')
+    list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    check_value_in_log_and_reset(topology_st, check_value=user_value)
+
+    log.info('Test dsidm user list with json')
+    args.json = True
+    list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    check_value_in_log_and_reset(topology_st, content_list=json_list, check_value=user_value)
```
Comment (droideck) [inline dirsrvtests/tests/suites/clu/dsidm_user_test.py:92]: I have an assertion error at that point:

```
INFO:dirsrvtests.tests.suites.clu.dsidm_user_test:Test dsidm user list with json
{
    "type": "list",
    "items": [
        "demo_user",
        "test_user_1000"
    ]
}
INFO:dirsrvtests.tests.suites.clu:Check if content is present in output
FAILED
```

Here:

```
    def check_value_in_log_and_reset(topology, content_list=None, content_list2=None, check_value=None,
                                     check_value_not=None):
        if content_list2 is not None:
            log.info('Check if content is present in output')
            for item in content_list + content_list2:
                assert topology.logcap.contains(item)

        if content_list is not None:
            log.info('Check if content is present in output')
            for item in content_list:
>               assert topology.logcap.contains(item)
E               AssertionError
```

**Review by droideck — CHANGES_REQUESTED** [review]

File: dirsrvtests/tests/suites/clu/dsidm_user_test.py (hunk 1 above)
Comment (bsimonova) [reply dirsrvtests/tests/suites/clu/dsidm_user_test.py:92]: Weird, it passes for me on Fedora, Rhel8.
On what version do you test? 
It is dependent on changes to lib389/cli_idm/__init__.py where I added printing the values to log.

Comment (droideck) [reply dirsrvtests/tests/suites/clu/dsidm_user_test.py:92]: Sorry for the fuss, it was my env issue :)

File: dirsrvtests/tests/suites/clu/dsidm_user_test.py [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+    check_value_in_log_and_reset(topology_st, content_list=user_content)
+
+    log.info('Test dsidm user get with json')
+    args.json = True
+    get(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    check_value_in_log_and_reset(topology_st, content_list=json_content)
+
+
+@pytest.mark.bz1893667
+@pytest.mark.skip("Will fail because of bz1893667")
```
Comment (droideck) [inline dirsrvtests/tests/suites/clu/dsidm_user_test.py]: Probably, `@pytest.mark.xfail(reason="") is a better option here.

**Review by droideck — APPROVED** [review]
Besides the small issue - LGTM! Ack:)

**Comment by vashirov** [comment]
This test fails with ERROR:
```
Error evaluating 'xfail' expression
    Will fail because of bz1893667
          ^
SyntaxError: invalid syntax
```
Looks like `pytest.mark.xfail` is missing `reason`.

File: src/lib389/lib389/cli_idm/__init__.py [hunk 3]
```diff
@@ -90,6 +94,8 @@ def _generic_get(inst, basedn, log, manager_class, selector, args=None):
     if args and args.json:
         o = mc.get(selector, json=True)
         print(o)
```
Comment (mreynolds389) [inline src/lib389/lib389/cli_idm/__init__.py:96]: With changes that were made in the last year, I think we can drop the "print" statements completely, and just use log.info().  Same for the changes below this...

Comment (bsimonova) [reply src/lib389/lib389/cli_idm/__init__.py:96]: Thanks for review, I am fixing it here: https://github.com/389ds/389-ds-base/pull/4594

**Comment by bsimonova** [comment]
@vashirov 

> This test fails with ERROR:
> 
> ```
> Error evaluating 'xfail' expression
>     Will fail because of bz1893667
>           ^
> SyntaxError: invalid syntax
> ```
> 
> Looks like `pytest.mark.xfail` is missing `reason`.

Sorry about that, I did not get a notification.
Fixing the issue here: https://github.com/389ds/389-ds-base/pull/4594

**Pull request merged**